packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1093,38 +1105,54 @@ def process_payload(self, payload: BaseTxPayload) -> None:
             )
 
         sender = payload.sender
-        if sender not in self.synchronized_data.participants:
+        if sender not in self.accepting_payloads_from:
             raise ABCIAppInternalError(
-                f"{sender} not in list of participants: {sorted(self.synchronized_data.participants)}"
+                f"{sender} not in list of participants: {sorted(self.accepting_payloads_from)}"
             )
 
         if sender in self.collection:
             raise ABCIAppInternalError(
                 f"sender {sender} has already sent value for round: {self.round_id}"
             )
 
+        if self._hash_length:
+            content = payload.data.get(self.payload_attribute)
+            if not content or len(content) % self._hash_length:
+                msg = f"Expecting serialized data of chunk size {self._hash_length}"
+                _logger.error(f"{msg}, got: {content} in {self.round_id}")
+                return
```

Comment:
you should `raise ABCIAppInternalError` here.

Reply:
as mentioned here: https://github.com/valory-xyz/open-autonomy/pull/983#discussion_r905565821

Reply:
And consistent with the other logic in this method

Reply:
I thought the idea was we should NOT do that because the content might come from a malicious agent, which could thus terminate the application https://github.com/valory-xyz/open-autonomy/pull/983#discussion_r905565813

Reply:
That's not what the comment says. The comment just explains the type or error.